### DISCUSSION TITLE
Tuning of the model parameters

### DISCUSSION THREAD

**Discussion by EddyG93** [NatLabRockies/floris#221] [Q&A] [answered]
Hi everyone, 

I am currently involved in a research project comparing the capabilities of different wake models implemented in FLORIS in terms of layout optimization. My reference layout will be the one of the Alpha Ventus Wind Farm and I have processed meteorological data from a neighbouring research platform (FINO1) available.
I am wondering if I can use the default model parameters  or if I have to tune them. In case of tuning do you have any ideas on how to start as there is no power data available?
Any help would be greatly appreciated!

Eddy

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Eddy,

Tuning of the model parameters in FLORIS remains a difficult and ambiguous task. We are currently working on this behind the scenes, but this is only its in preliminary stages. Previously, we have tuned FLORIS to high-fidelity simulation data (SOWFA) comparing flow fields and power productions. Also, in field experiments, we may tune certain parameters by hand to match the energy ratios better.

Here is some literature you may find useful:
- Section 2.3 in ["Improving wind farm flow models by learning from operational data", Schreiber et al., 2020](https://wes.copernicus.org/articles/5/647/2020/)
This paper describes both real-time model corrections and a baseline tuning of the model parameters.
- Section 3 in[ "Closed-loop model-based wind farm control using FLORIS under time-varying inflow conditions
", Doekemeijer et al., 2020](https://pure.tudelft.nl/ws/portalfiles/portal/72881357/1_s2.0_S0960148120305358_main.pdf)
Where I tune a MATLAB-equivalent of FLORIS to SOWFA data. The method is identical, but I will stress that parameters may have a different interpretation between the MATLAB and NRELs Python model. Do not interchangeably use these models.
- And here's a paper explaining energy ratios, which we may use for tuning selective parameters by hand, ["Continued results from a field campaign of wake steering – Part 2", Fleming et al., 2020](https://www.nrel.gov/docs/fy20osti/77681.pdf)]

Generally, though, if you do not have any or only little data available, then I suggest you stick to the default model parameters. In our experience, They give a good match with SCADA data from offshore and flat-terrain onshore wind farms.

Bart

**Reply by EddyG93** [reply]
Hello Bart, 

thanks a lot for your answer. The tuning seems to be a quite ambitious task. Having read through the suggested literature and considering the fact that there is barely data available I decided to follow your suggestion to stick to the default model parameters. 
Can you recomend any literature material supporting your statement that the default parameters match well with offshore SCADA data? According to my literature study there aren't to many field experiments. 
As you mentioned before, there is a lot of supportive work showing that the models match well with SOFWA simulations. However, the default model parameters in the current version of FLORIS partly differ from those values used in the simulations and from the orginal models which are presented in the FLORIS documentation. Can you provide any reference for the current default parameters?

Eddy

**Reply by Bartdoekemeijer** [reply]
Hey Eddy,

I am currently in the process of writing a publication where I validate FLORIS with historical SCADA data from 3 different offshore wind farms. I am not able to share any results of that currently, but hoping to submit the manuscript within the next 2 months. At that time, I can also share a preprint with you.

Bart

**Reply by mr483** [reply]
Hi @Bartdoekemeijer, 

I noticed your paper is referenced in the latest published CC paper ("Addressing deep array effects..."), but that it is still under review. Is there any chance you could share a copy of the preprint? Thanks!

**Reply by bayc** [reply]
@mr483 The paper preprint can be found here: https://wes.copernicus.org/preprints/wes-2022-17/.

**Reply by mr483** [reply]
Sorry, I was actually referring to the paper that is cited in the latest CC paper, with the citation: 

Doekemeijer, B. M., Simley, E., and Fleming, P. A.: Comparison of the Gaussian wind farm model with historical data of three offshore wind farms, Energies special Issue: “Fast-running Engineering Models of Wind Farm Flows”, submitted, in review, 2022.

**Reply by Bartdoekemeijer** [reply]
Hi @mr483 . The paper was formally accepted by the editors and I'm submitting the final proofread/approval today. The official paper should be publicly available not too long from now, hopefully this week.

**Reply by mr483** [reply]
Great, thank you!

**Reply by Bartdoekemeijer** [reply]
@mr483 you can find the article here: https://www.mdpi.com/1996-1073/15/6/1964.

**Comment by Bartdoekemeijer** [comment]
Hi Eddy. Please see my reply to #276 for some preliminary validation results of FLORIS with historical data. We expect to be able to share the scientific articles we have been preparing over the next couple months.
